testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/TestRun.java

```diff
@@ -251,6 +256,13 @@ private Environment startEnvironment()
                 testsContainer.start();
             }
 
+            String grepConnectorsOutput = coordinatorContainer.execCommand("/usr/bin/grep", "--recursive", "--fixed-strings", "connector.name", "/docker/presto-product-tests/conf/presto/etc/catalog");
```

Comment:
Use `TABLE system.metadata.catalogs` instead of ruthless grepping.

Reply:
I was hoping to detect not only connectors/catalogs, but also things like access control factories. 
I'm trying to refine the grep pattern to include those as well, but I don't think there's a corresponding Trino table for that.
Are we sure we want to limit ourselves to connectors only? It will mean missing out on some optimization opportunities.

Reply:
To map connectors to Maven modules, we'd need to associate them with the plugin class name, like `jmx` -> `io.trino.plugin.jmx.JmxPlugin`. 

This is only available in logs now, but I don't want to rely on parsing unstructured text:
```
2022-01-21T11:06:57.919Z        INFO    main    io.trino.server.PluginManager   -- Loading plugin /data/trino/plugin/jmx --
2022-01-21T11:06:57.925Z        INFO    main    io.trino.server.PluginManager   Installing io.trino.plugin.jmx.JmxPlugin
2022-01-21T11:06:57.931Z        INFO    main    io.trino.server.PluginManager   Registering connector jmx
2022-01-21T11:06:57.932Z        INFO    main    io.trino.server.PluginManager   -- Finished loading plugin /data/trino/plugin/jmx --
```

I can suggest three solutions:
1. Add a new table, like `system.metadata.plugins` with a list of installed plugins, and their features.
2. Add some main class to the `PluginManager` that could be executed via CLI and print info about all plugins found, that Trino would try to load.
3. Add a list of features inside the plugin's service jar, similar to `META-INF/services/io.trino.spi.Plugin` - this sounds easiest but requires changes in all plugins and there's a risk that some plugins might not include it or the list might not be up to date

Reply:
Ad 1. https://github.com/trinodb/trino/pull/10830